cypress/page/dresses-list.page.ts

```diff
@@ -0,0 +1,27 @@
+class DressesListPage {
+
+    private dressItem: string;
+    private dressName: string;
+  
+    constructor(){
+      this.dressItem = ".product-container"
+      this.dressName = `${this.dressItem} .product-name`
+    }
+  
+    getDressProducts(){
+      return cy.get(this.dressItem)
+    }
+  
+    validateItemsNumber(itemsNumber: number){
+      cy.get(this.dressItem).should("have.length", itemsNumber)
+    }
+  
+    validateItemsNames(names: string[]){
+      cy.get(this.dressName).each((item, index) => {
+        cy.wrap(item).should("contain.text", names[index])
```

Comment:
Here is a better way, obtain the text that is embedded in the element, trim it as a string and now you will not need the \t\n.. that you have in the test data